cmd/server.go

```diff
@@ -10,14 +10,15 @@ import (
 	"github.com/spf13/cobra"
 )
 
+var ctx = context.Background()
+var out = new(bytes.Buffer)
+var forceFlag bool
```

Comment:
Why are these global? Is this needed (or idiomatic?)

Reply:
I need to access in my commands, had them in one scope only which was very bad

Reply:
Ah, well, `forceFlag` I can understand why this is global and this is okay. But why are `ctx` and `out` buffer global? Is this needed?

Reply:
Ctx yes for the new subcommands since they need this parameter, out no, but as I was moving ctx out I thought maybe there's an future need  for out as well

Reply:
Understand. Not a 100% Go-pro here myself but in most languages you want to limit the scope of a variable (esp. if it is mutable) as much as possible. So the smaller the scope a variable is visible (or addressable) the better.

The declaration's lexical block determines its scope, which right now is at the package level for `out` and `ctx`. So it is addressable / visible at package level, meaning anything in the package (so even other .go files in the same package) can see and change that variable, making reasoning much harder.

Try to limit the scope of `ctx` and `out` to the functions or lexical blocks where they are actually used is the right thing to do here.

Reply:
@bkircher is right about the scope of the `context`. The `context` is supposed to flow like a waterfall. That means you should pass it function to function, as it controls the whole control "tree".

Reply:
Btw, the subcommands are irrelevant (they cannot be executed at the same time because of the `if-else` statement). So that they should use the different `context`. You'd better insert context declaration line on line 60, 68 like the line 18 which you removed (https://github.com/gridscale/gscloud/pull/32/files/1bf3e2b208e64f2abdbb6141c7b343ab73c2dc9e#diff-1427b112aa72f64adcf839007e954d56L18)

Reply:
I made the changes, but i don't understand why you can't just have one `context`, all functions use the same anyways, isn't that redundancy in the code?

Reply:
I think context is a handle (like a cancellation token, if you heard of that before) that callers can use to cancel or otherwise influence concurrent operations. So I think that you want to use a different `Context` object for each operation.

Good read: https://blog.golang.org/context